lib/core/elastic_connector_actions.rb

```diff
@@ -97,6 +98,15 @@ def client
         @client ||= Utility::EsClient.new
       end
 
+      # should only be used in CLI
+      def ensure_content_index_exists(index_name, use_icu_locale = false, language_code = nil)
+        settings = Utility::Elasticsearch::Index::TextAnalysisSettings.new(:language_code => language_code, :analysis_icu => use_icu_locale).to_h
+        mappings = Utility::Elasticsearch::Index::Mappings.default_text_fields_mappings(:connectors_index => true)
+
+        body_payload = { settings: settings, mappings: mappings }
+        ensure_index_exists(index_name, body_payload)
```

Comment:
It will only try to create the index when it doesn't exist.
https://github.com/elastic/connectors/blob/75eb472237c206d3496206679b21780a59504130/lib/core/elastic_connector_actions.rb#L122

>if the index already exists but without mappings

I think this makes sense. This is the expected case when operator creates the index in UI (index created without mapping). Connector service is supposed to update the mappings. @mchernyavskaya , I think `ensure_content_index_exists` should handle this:
1. If index does not exist, create the index with mappings
2. If index exists without mappings, update the mappings
3. Else, do nothing.

And the above should be done in both `console_app` and `app`.

Reply:
I was wondering if you already discussed the possibility of creating indices with the App search mappings from Kibana? I assume Kibana can also set settings/mappings to the index.

Reply:
>I assume Kibana can also set settings/mappings to the index.

No, Kibana will only create the index, without mappings. Only connector service will know the mappings for each connector (e.g. `GitLab`)

But for system indices (e.g. `.elastic-connectors`), both Kibana and connector service should create them with mappings.

Reply:
Thanks @wangch079. Then all you mentioned [here](https://github.com/elastic/connectors/pull/172#discussion_r918598703) make sense to me.

Reply:
@wangch079 @vidok how do we know if mappings exist? there's always **some** mappings - even dynamic, if there's at least one record. So my question is:
1. How do we know that mappings haven't been created?
2. How do we know that mappings are **wrong**?

Reply:
> there's always **some** mappings - even dynamic

This is not true. Kibana will only create the index with the index name, so the mapping will be empty.

Try to run 
```
PUT stub-connector
```

And run
```
GET stub-connector/_mapping
```

You will get
```
{
  "stub-connector": {
    "mappings": {}
  }
}
```

>How do we know that mappings are wrong?

Kibana will do the validation before creating the index, so we can assume the index is created by Kibana, not an existing index. If the mapping in the index is different from what we have in connector service, it means the mappings in connector service are updated. I'm not sure how we handle connector mappings update (cc @danajuratoni ), as [Elasticsearch itself doesn't allow changing the mapping or field type of an existing field](https://www.elastic.co/guide/en/elasticsearch/reference/current/indices-put-mapping.html#updating-field-mappings).

But I don't think we should worry about this in 8.4

Reply:
> I think this functionality can be a part of `connectors_utility` gem.

I think this functionality might be out of scope for this PR in that case.

Reply:
>So if there's already something in the index, the mapping isn't empty.

This makes sense. But we should have this `ensure_mappings_updated` before any data can be ingested from connector.

>I think this functionality might be out of scope for this PR in that case.

Agree that we don't need to worry about mappings compatibility. But we do need to have mappings updated by connector service. This is the core requirement of https://github.com/elastic/enterprise-search-team/issues/2127